src/python/views/demo_tigergraph_circle/__init__.py

```diff
@@ -0,0 +1,774 @@
+def run_filters(user_id, conn):
+    global metrics
+    # global conn
+
+    # if conn is None:
+    #     conn = tg.TigerGraphConnection(host="https://fraud-streamlit.i.tgcloud.io", graphname="AntiFraud",
+    #                                    username="tigergraph", password="tigergraph")
+
+    # conn.getToken("tufp2os5skgljafj7ol4ikht2atc7rbj")
+    logger.info("Installing Queries")
+    res = conn.gsql(
+    '''
+    use graph {}
+    ls 
+    '''.format(conn.graphname), options=[])
+
+    ind = res.index('Queries:') + 1
+    installTran = True
+    for query in res[ind:]:
+        if 'totalTransaction' in query:
+            installTran = False
+    
+    if installTran:
+        conn.gsql(
+        '''
+        use graph AntiFraud
+        CREATE QUERY totalTransaction(Vertex<User> Source) FOR GRAPH AntiFraud {  
+            start = {Source};
+
+            transfer = SELECT tgt
+                FROM start:s -(User_Transfer_Transaction:e) - :tgt;
+
+            receive = select tgt
+                FROM start:s -(User_Recieve_Transaction:e) -:tgt;
+
+            PRINT transfer, receive;
+        }
+        Install query totalTransaction 
+        ''', options=[])
+
+    logger.info('Querying Tigergraph')
+    tic = time.perf_counter()
+
+    results = conn.runInstalledQuery("circleDetection", {"srcId": user_id}, sizeLimit=1000000000, timeout=120000)
+    results = results[0]['@@circleEdgeTuples']
+
+    out = []
+    from_ids = []
+    to_ids = []
+    times = []
+    amounts = []
+    types = []
+    from_types = []
+    to_types = []
+
+    for o in results:
+        for s in o:
+            if {"from_id": s["e"]["from_id"], "to_id": s["e"]["to_id"], "amount": s["amount"], "time": s["ts"],
+                "type": s["e"]["e_type"]} not in out:
+                out.append(
+                    {"from_id": s["e"]["from_id"], "to_id": s["e"]["to_id"], "amount": s["amount"], "time": s["ts"],
+                     "type": s["e"]["e_type"]})
+                from_ids.append(s["e"]["from_id"])
+                to_ids.append(s["e"]["to_id"])
+                amounts.append(s["amount"])
+                times.append(s["ts"])
+                types.append(s["e"]["e_type"])
+                from_types.append(s['e']['from_type'])
+                to_types.append(s['e']['from_type'])
+
+    ############# BEGIN
+    edges_df = pd.DataFrame({
+        'from_id': from_ids,
+        'to_id': to_ids,
+        'amount': amounts,
+        'time': times,
+        'type': types
+    })
+    node_idf = []
+    typef = []
+    for i in range(len(from_ids)):
+        if from_ids[i] not in node_idf:
+            node_idf.append(from_ids[i])
+            typef.append(from_types[i])
+        if to_ids[i] not in node_idf:
+            node_idf.append(to_ids[i])
+            typef.append(to_types[i])
+
+    nodeType2color = {
+        'User': 0xFF740000,  # orange
+        'Transaction': 0xA5A5A500  # light gray
+    }
+
+    edgeType2color = {
```

Comment:
270-290 would probably be simpler and more pedagogic  as doing in the `plot_ur()` call as a couple one-liners:

`g = g.encode_point_color('type', categorical_mapping={'User': 'orange', 'Transaction': '#CCC'})`

Reply:
Changed the color settings to categorical mapping as suggested